compiler/vectorized_test.go

```diff
@@ -208,6 +208,89 @@ func TestVectorizedFns(t *testing.T) {
 			vectorizable: true,
 			skipComp:     true,
 		},
+		{
+			name:         "logical expressions",
+			fn:           "(r) => ({ r with c: r._a and r._b, d: r._a or r._b })",
+			vectorizable: true,
+			skipComp:     true,
+		},
+	}
+
+	logicalTests := []struct {
```

Comment:
These go-side tests are the only place we verify that values in the vectorized code path are retained/released correctly. It appears we didn't have any for logical ops before now.

It looks like I'm testing for correctness here, but that's the aim. Mostly I'm trying to hit different short-circuit cases to verify memory accounting when (main) early returns are in play.